packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -888,6 +902,12 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
       engineVersion: props.engine.engineVersion?.fullVersion,
       licenseModel: props.licenseModel,
       timezone: props.timezone,
+      ...props.parameters && {
+        dbParameterGroupName: new ParameterGroup(this, 'ParameterGroup', {
+          engine: props.engine,
+          parameters: props.parameters,
+        }).bindToInstance({}).parameterGroupName,
+      },
```

Comment:
Can we extract this all of this to a `parameterGroupName: string | undefined` local variable? This code is way, way too clever for my liking.